docs/source/extending.rst

```diff
@@ -0,0 +1,103 @@
+.. _extending:
+
+
+.. note::
+
+   Make sure you've read the `design section <design>`_ of the documentation,
+   otherwise the rationale behind extending ibis won't make much sense.
+
+
+Extending Ibis
+==============
+
+There are two integration points that typical users want to extend:
+
+#. Additional operations
+#. Support for a new backend
+
+Adding a new operation (``Node`` subclass)
+------------------------------------------
+
+Let's go through adding a `sha1`_ method to ibis, implemented in the BigQuery
```

Comment:
ref to BQ (maybe doc-section in ibis)

Reply:
I'll link to the sha1 wikipedia article here https://en.wikipedia.org/wiki/SHA-1